hathor/p2p/manager.py

```diff
@@ -302,10 +303,32 @@ def has_synced_peer(self) -> bool:
         for conn in connections:
             assert conn.state is not None
             assert isinstance(conn.state, ReadyState)
-            if conn.state.is_synced():
+            if conn.is_sync_enabled() and conn.state.is_synced():
                 return True
         return False
 
+    def get_best_blockchain_height_among_peers(self) -> Optional[HeightInfo]:
+        """ Return the best blockchain height among all peers.
+
+        Note that not necessarily they all have the same best blockchain,
+        so use this method with caution. We only consider the one with the
+        highest height regardless of whether the blockchains are the same.
+        """
+        connections = list(self.iter_ready_connections())
+        best_height_info = None
+
+        for conn in connections:
+            assert conn.state is not None
+            assert isinstance(conn.state, ReadyState)
+            peer_best_blockchain = list(sorted(conn.state.peer_best_blockchain, key=lambda x: x.height, reverse=True))
```

Comment:
Usually the `peer_best_blockchain` will come already sorted in descending order. If this sorting is already optimized, then the time complexity would be `O(n)`, assuming `n` as the number of elements in the `peer_best_blockchain`. Given `n` always equal `10`, it yields `O(1)` for this instruction. But everything depends on the `sorted` implementation.